packages/itwinui-react/src/core/ThemeProvider/ThemeProvider.tsx

```diff
@@ -96,19 +115,23 @@ export const ThemeProvider = React.forwardRef((props, forwardedRef) => {
     theme: themeProp = 'inherit',
     children,
     themeOptions,
+    portalContainer: portalContainerProp,
     ...rest
   } = props;
 
-  const portalContainerRef = React.useRef<HTMLDivElement>(null);
+  const [portalContainerState, setPortalContainerState] = React.useState(
+    portalContainerProp || null,
+  );
+  const portalContainer = portalContainerProp || portalContainerState;
```

Comment:
What is the need for `portalContainer`? I am a bit confused why `portalContainerState` is not enough.

Reply:
We want the prop to take preference. The state should only be used when the prop is not passed.

Reply:
i've created a new `useUncontrolledState` hook to make this less confusing